src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/ConsulWatcherTest.groovy

```diff
@@ -270,33 +315,25 @@ class ConsulWatcherTest extends Specification {
 
     def "should not invoke callbacks for disposed watch"() {
         given:
-        def first = 0
-        def second = 0
+        def counter = 0
 
         consul.stubFor(get(urlPathEqualTo('/cancel'))
                 .willReturn(aResponse()
-                .withHeader('X-Consul-Index', '123')
-                .withFixedDelay(100)
-                .withBody('123')))
+                        .withHeader('X-Consul-Index', '123')
+                        .withFixedDelay(100)
+                        .withBody('123')))
 
         when:
         def canceller = watcher
-                .watchEndpoint("/cancel", { it -> first += 1 }, { it -> first += 1 })
-
-        def canceller2 = watcher
-                .watchEndpoint("/cancel", { it -> second += 1}, { it -> second += 1})
+                .watchEndpoint("/cancel", { it -> counter += 1}, { it -> counter += 1 })
 
+        def beforeCancel = counter
         canceller.cancel()
+        Thread.sleep(2000)
```

Comment:
can you comment on this `sleep`, what's the race it is addressing?

Reply:
So maybe use ` await()` with `.isCancelled()`?

Reply:
i changed it